_resources/port1.0/group/cmake-1.1.tcl

```diff
@@ -118,11 +119,16 @@ proc cmake::handle_generator {option action args} {
                 destroot.env-append DESTDIR=${destroot}
             }
             default {
+                if {[llength ${args}] > 1 || [llength ${args}] <= 0} {
```

Comment:
As @larryv pointed out, this is equal to `[llength $args] != 1`. Also not that my suggestion was doing this above the switch and then running the switch only on the first index in this list, i.e.

```tcl
if {$action eq "set"} {
    if {[llength $args] != 1} {
        error "cmake.generator requires a single value (not '$args')"
    }
    switch -glob [lindex $args 0] {
        # ...
    }
}
```

Otherwise, you'll still have the same list issues as before.